### PR TITLE
Thimble issue723

### PR DESCRIPTION
**Pull request by humphd**
This is a WIP, but I have a lot of it working now.  I still need to refactor the DnD code a bit more.  I'll finish it tomorrow.

### CODE REVIEW

**Comment by humphd** [comment]
This is ready for review.  To test it, open your console and type:

```js
bramble.showUploadFilesDialog()
```

I debated refactoring the Selfie stuff more, and making it into its own extension, but decided against it.  I've also re-used the internal Drag and Drop stuff, which I mostly rewrote for Bramble.

File: src/extensions/default/UploadFiles/UploadFilesDialog.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    };
+    FileUploadDialog.prototype._handleFromComputer = function() {
+        var self = this;
+        var deferred = self.deferred;
+
+        self.hide();
+
+        function _processFiles(e) {
+            var files = self.fileInput.getFiles();
+            console.log('files', files);
```
Comment (gideonthomas) [inline src/extensions/default/UploadFiles/UploadFilesDialog.js]: leftover log

File: src/extensions/default/UploadFiles/main.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    function showUploadFiles() {
+        return UploadFilesDialog.show();
+    }
+
+    function addCommand() {
+        CommandManager.register(CMD_UPLOAD_FILES_TEXT, CMD_UPLOAD_FILES_ID, showUploadFiles);
+    }
+
+	ExtensionUtils.loadStyleSheet(module, "styles.less");
```
Comment (gideonthomas) [inline src/extensions/default/UploadFiles/main.js]: nit: dedent

**Comment by gideonthomas** [comment]
r=me so far

**Comment by sedge** [comment]
This sucking in the random files I dragged from my desktop, unzipping as it went, was a thing of beauty. r+ from testing.

Maybe get @gideonthomas to review the code?

**Comment by gideonthomas** [comment]
r=me

**Pull request merged**